pyproject.toml

```diff
@@ -3,7 +3,7 @@ requires = [
     "setuptools>=67.8",
     "wheel",
     "cython~=3.0.2",
-    "numpy==2.0.0rc1"
+    "numpy~=2.0"
```

Comment:
```suggestion
    "numpy>=2"
```
I only suggest this because the pin in [setup.py](https://github.com/rasterio/rasterio/blob/f3018c78a4fe06863fdadf23536564c16dec2ced/setup.py#L260) is open. It prevents having a build version less than the runtime version.

Reply:
@snowman2 I'm missing something. How does it prevent having a build version less than the runtime version?

Reply:
When numpy 3 is released, rasterio with build with numpy 2 and run with 3. Probably won't be an issue for a while 🤷‍♂️